guides/source/active_support_instrumentation.md

```diff
@@ -3,623 +3,1160 @@
+The event is emitted when a job is enqueued to run as soon as possible.
+
+For example:
+
+```ruby
+MyJob.perform_later("now")
+```
+
+The event payload (`event.payload`) includes the following keys (with typical
+example values).
+
+| Payload Key | Description                     | Example        |
+| ----------- | ------------------------------- | -------------- |
+| `:adapter`  | QueueAdapter processing the job | `#<...>`       |
+| `:job`      | Job object                      | `#<MyJob ...>` |
 
 #### `enqueue_retry.active_job`
 
-| Key          | Value                                  |
-| ------------ | -------------------------------------- |
-| `:job`       | Job object                             |
-| `:adapter`   | QueueAdapter object processing the job |
-| `:error`     | The error that caused the retry        |
-| `:wait`      | The delay of the retry                 |
+The event is emitted when a job is scheduled for retry due to an error.
+
+For example:
+
+```ruby
+MyJob.perform_later("fail to retry")
+# inside perform, raise Timeout::Error => Active Job schedules retry
+```
+
+The event payload (`event.payload`) includes the following keys (with typical
+example values).
+
+| Payload Key | Description                     | Example                 |
+| ----------- | ------------------------------- | ----------------------- |
+| `:job`      | Job object                      | `#<MyJob ...>`          |
+| `:adapter`  | QueueAdapter processing the job | `#<...>`                |
+| `:error`    | The error that caused the retry | `#<Timeout::Error ...>` |
+| `:wait`     | The delay of the retry          | `30.seconds`            |
 
 #### `enqueue_all.active_job`
 
-| Key          | Value                                  |
-| ------------ | -------------------------------------- |
-| `:adapter`   | QueueAdapter object processing the job |
-| `:jobs`      | An array of Job objects                |
+The event is emitted when multiple jobs are enqueued together.
+
+For example:
+
+```ruby
+jobs = [ MyJob.new, MyJob.new ]
+ActiveJob::Base.enqueue_all(jobs) # when supported by the adapter
+```
+
+The event payload (`event.payload`) includes the following keys (with typical
+example values).
+
+| Payload Key | Description                     | Example                 |
+| ----------- | ------------------------------- | ----------------------- |
+| `:adapter`  | QueueAdapter processing the job | `#<...>`                |
+| `:jobs`     | An array of Job objects         | `[ #<MyJob ...>, ... ]` |
 
 #### `perform_start.active_job`
 
-| Key          | Value                                  |
-| ------------ | -------------------------------------- |
-| `:adapter`   | QueueAdapter object processing the job |
-| `:job`       | Job object                             |
+The event is emitted when job execution starts (on the worker).
+
+For example:
+
+```ruby
+# Fired on the worker right before perform begins
```

Comment:
I think we can show `perform_now` here as `_perform_job` [emits it](https://github.com/rails/rails/blob/cf4e3c508adf930eb70e4b3e6e16d2ca83b7e9b3/activejob/lib/active_job/instrumentation.rb#L43).
```suggestion
MyJob.perform_now # Fired on the worker right before perform begins
```